packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,493 @@
+
+    List<TreemapNode> list2 = [];
+    int list2Size = 0;
+    List<TreemapNode> list3 = [];
+    int list3Size = 0;
+
+    // The amount of data we have from pivot + 1 (exclusive)
+    // In another words, if we only put one data in l2, how many are left for l3?
+    // [L1, pivotIndex, data, |d|] d = 2
+    final int l3MaxLength = children.length - pivotIndex - 1;
+    int bestIndex = 0;
+    double pivotBestWidth = 0;
+    double pivotBestHeight = 0;
+
+    // We need to be able to put at least 3 elements in l3 for this algorithm.
+    if (l3MaxLength >= 3) {
+      double pBestAspectRatio = double.infinity;
+      // Iterate through different combinations of list2 and list3 to find
+      // the combination where the aspect ratio of pivot is the lowest.
+      for (int i = pivotIndex + 1; i < children.length; i++) {
+        final int list2Size = computeSize(children, pivotIndex + 1, i);
+
+        // Calculate the aspect ratio for the pivot treemap node.
+        final double pAndList2Ratio = (pSize + list2Size) / totalSize;
+        final double pRatio = pSize / (pSize + list2Size);
+
+        final double pWidth =
+            isHorizontalRectangle ? pAndList2Ratio * width : pRatio * width;
+
+        final double pHeight =
+            isHorizontalRectangle ? pRatio * height : pAndList2Ratio * height;
+
+        final double pAspectRatio = pWidth / pHeight;
+
+        // Best aspect ratio that is the closest to 1.
+        if ((1 - pAspectRatio).abs() < (1 - pBestAspectRatio).abs()) {
+          pBestAspectRatio = pAspectRatio;
+          bestIndex = i;
+          // Kept track of width and height to construct the pivot cell.
+          pivotBestWidth = pWidth;
+          pivotBestHeight = pHeight;
+        }
+      }
+      // Split the rest of the data into list2 and list3
+      // [L1, pivotIndex, [L2 bestIndex], L3]
+      list2 = children.sublist(pivotIndex + 1, bestIndex + 1);
+      list2Size = computeSize(list2, 0, list2.length - 1);
+
+      list3 = children.sublist(bestIndex + 1);
+      list3Size = computeSize(list3, 0, list3.length - 1);
+    } else {
+      // Put all data in l2 and none in l3.
+      list2 = children.sublist(pivotIndex + 1);
+      list2Size = computeSize(list2, 0, list2.length - 1);
+
+      final double pdAndList2Ratio = (pSize + list2Size) / totalSize;
+      final double pdRatio = pSize / (pSize + list2Size);
+      pivotBestWidth =
+          isHorizontalRectangle ? pdAndList2Ratio * width : pdRatio * width;
+      pivotBestHeight =
+          isHorizontalRectangle ? pdRatio * height : pdAndList2Ratio * height;
+    }
+
+    // Contruct list 1 sub-treemap.
+    final double list1SizeRatio = list1Size / totalSize;
+    final double list1Width =
+        isHorizontalRectangle ? width * list1SizeRatio : width;
+    final double list1Height =
+        isHorizontalRectangle ? height : height * list1SizeRatio;
+    List<Cell> list1Cells;
+    list1Cells = buildTreemap(
```

Comment:
create a new TreeMapWidget instead of using the method `buildTreemap`. Here and for the other cases in this method